File: Codeforces/codeTon3/C.py

for _ in range(int(input())):
    n = int(input())
    a = list(map(int, input()))
    b = list(map(int, input()))
    ok = True
    ans = []
    invs = 0
    if a != b:
        for i in range(n):
            if a[i] != 1 - b[i]:
                ok = False
                break
    else:
        ans.append((1, n))
        for i in range(n):
            a[i] = 1 - a[i]
    if not ok:
        print('NO')
        continue
    print('YES')
    for i in range(n):
        if a[i] == 1:
            ans.append((i + 1, i + 1))
            invs += 1
    if invs % 2 == 0:
        ans.append((1, n))
        ans.append((1, 1))
        ans.append((2, n))
    print(len(ans))
    for i in ans:
        print(*i)

File: Codeforces/codeTon3/C.cpp

#include <iostream>
#include <string>
#include <vector>

using namespace::std;

int main()
{
    int t;
    cin >> t;
    while (t--) {
        int n;
        string a, b;
        cin >> n >> a >> b;
        for (int i=0; i < n; ++i) {
            a[i] = a[i] - '0';
            b[i] = b[i] - '0';
        }
        bool ok = true;
        vector<int> ans;
        ans.reserve(n + 10);
        int invs = 0;
        if (a != b) {
            for (int i=0; i < n; ++i) {
                if (a[i] != 1 - b[i]) {
                    ok = false;
                    break ;
                }
            }
        } else {
            ans.push_back(1);
            ans.push_back(n);
            for (int i=0; i < n; ++i) {
                a[i] = 1 - a[i];
            }
        }
        if (not ok) {
            cout << "NO\n";
            continue ;
        }
        cout << "YES\n";
        for (int i=0; i < n; ++i) {
            if (a[i] == 1) {
                ans.push_back(i + 1);
                ans.push_back(i + 1);
                ++invs;
            }
        }
        if (invs % 2 == 0) {
            ans.push_back(1);
            ans.push_back(n);
            ans.push_back(1);
            ans.push_back(1);
            ans.push_back(2);
            ans.push_back(n);
        }
        cout << ans.size() / 2 << endl;
        for (int i=0; i < ans.size(); i += 2) {
            cout << ans[i] << ' ' << ans[i + 1] << endl;
        }
    }
}
